js/src/api/contract/contract.js

```diff
@@ -218,14 +218,19 @@ export default class Contract {
   }
 
   _encodeOptions (func, options, values) {
-    options.data = this.getCallData(func, options, values);
-    return options;
+    const data = this.getCallData(func, options, values);
```

Comment:
Since we fixed a bug, would love a testcase that reproduces it and verifies it fixed so it doesn't happen again. (In the cas eof fixing bugs, should get into that habit overall)